### PR TITLE
Features/choo choo calc

### PR DESCRIPTION
**Pull request by zbrotherton2158**
# Pull Request Details

## Which does this affect?

Changes arm to use choo choo encoder for feedback

## Checklist

- [x] My code follows the code style of this project.
- [ ] My code has been tested on the Robot OR includes simulation code to prove functionality.

### CODE REVIEW

File: src/main/java/frc/robot/classes/ChooChooMap.java [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+    public double toChooChooAngle(double armAngle){
+        return chooChooMap.get(truncate(armAngle, 1));
+    }
+
+    private double truncate(double a, int places){
+        return Math.round(a * Math.pow(10, places)) / Math.pow(10, places);
+    }
+
+    public ChooChooMap(){
```
Comment (azvanderpas) [inline src/main/java/frc/robot/classes/ChooChooMap.java:16]: Include the code used to generate this list (somewhere outside the src directory so gradle doesn't find)
Add comments to what this class is (desmos links, how it was calculated, etc.

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 2]
```diff
@@ -217,15 +220,19 @@ public ArmSubsystem() {
 
   //Returns sim encoder position (No offset) if in simulation, the actual position otherwise
   public double getBaseAngle() {
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/ArmSubsystem.java:222]: Check references to this method, and check whether they should be switched to getChooChooAngle

**Review by azvanderpas — CHANGES_REQUESTED** [review]
See other comments in #62 as well

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 3]
```diff
@@ -313,7 +320,7 @@ public void setDesiredPositions(double x, double y) {
       desiredYPosition.setDouble(y);
     }
     
-    setBaseReference(desiredBaseAngle);
+    setBaseReference(chooChooMap.toChooChooAngle(desiredBaseAngle));
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/ArmSubsystem.java:323]: desiredBaseAngle is in radians, needs to be converted to degrees to lookup in the HashMap

**Comment by zbrotherton2158** [comment]
Not necessary, should be able to continue using the arm encoder as feedback

**Pull request closed without merging**